libvast/vast/data.hpp

```diff
@@ -314,6 +313,18 @@ bool convert(const data& xs, json& j);
 /// data.
 bool convert(const data& x, json& j, const type& t);
 
+// -- YAML -------------------------------------------------------------
+
+/// Parses YAML into a data.
+/// @param str The string containing the YAML content
+/// @returns The parsed YAML as data.
+caf::expected<data> from_yaml(std::string_view str);
+
+/// Prints data as YAML.
+/// @param x The data instance.
+/// @returns The YAML representation of *x*.
```

Comment:
```suggestion
/// @returns The YAML representation of x on success, or an error.
```

If you want to use highlighting, the proper way to do it for Doxygen / standardese would be `\p x` if I recall correctly. However, since we're not doing this anywhere else, let's just not do it here either.